sourcegraph/sourcegraph
# docs: rework setup instructions

While we were pairing with @mrnugget on `sg setup`, it became clear that the current documentation is quite confusing for newcomers. 

This set of changes reorganise the getting started guide for local development. 

Goals:

- simplify those pages which have became hairy over time
- provide a place to put alternative setup instructions

Non-goals:

- reorganising the development section

Notable changes:

- Local Env is now a root section
  - mixing development how-tos with local env how-tos is confusing and contributed to a sense of mess for newcomers
  - a clear separation between the two will make upcoming contributions much easier.
- A page listing alternative options to install databases and languages have been added
- A ton of links have been updated and many links in the handbook will need to be updated.

How to review this:

- check out the branch locally
- `sg run docsite` 
- browse the new section

<details>
  <summary>Screenshot</summary>
  
<img width="1551" alt="CleanShot 2021-11-20 at 00 02 36@2x" src="https://user-images.githubusercontent.com/10151/142702030-fe3b6736-1c6a-4e2c-93bc-e47a8010a94d.png">

</details>
